viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
+output: the average rating
+# {   'watched': [   {   'genre': 'Fantasy',
+#                        'rating': 4.8,
+#                        'title': 'The Lord of the Functions: The Fellowship of '
+#                                 'the Function'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Two '
+#                                 'Parameters'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Return of the '
+#                                 'Value'},
+#                    {   'genre': 'Action',
+#                        'rating': 2.2,
+#                        'title': 'The JavaScript and the React'},
+#                    {'genre': 'Intrigue', 'rating': 2.0, 'title': 'Recursion'},
+#                    {   'genre': 'Intrigue',
+#                        'rating': 4.5,
+#                        'title': 'Instructor Student TA Manager'}]}
+'''
+
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"] ## array of dictionaries
+    total_rating = []
+    counter = 0 
+## for a dictionary in the list of movies
+    for movie_data in watched_movies:
+        v = movie_data["rating"]
+        total_rating.append(v)
+    if len(total_rating) == 0:
+        return 0.0
+    average = sum(total_rating) / len(total_rating)
+    return average
+
+
+
+## def function to track most watched genre
+## COUNT of times genre appears
+## If there is no amount, return error message
+# '''
+# input: take in the parameter 'user_data' : a nested dictionary 
+# with a VALUE 'watched' list of movie dictionaries
+# each movie has the KEY "genre"
+# >> this VALUE represents user's list of watched movies
+# >> each watched movie contains a genre
+# >>>> the VALUES of  KEY "genre" is a string
+# create a COUNTER to count which "genre" occurs most often
+# If the value of "watched" is empty, return None
 
+# output: The string of the genre that is most frequently watched,
+# if value of watched is an empty list could try ... else none
 
+# '''
+
+def get_most_watched_genre(user_data):
+    watched_movies = user_data["watched"]
+    popular_genre = []
+    count = 0
+
+    if watched_movies == []:
+        return None
+    for movie_data in watched_movies: 
+        popular_genre.append(movie_data["genre"])
+    for movie_data in watched_movies: 
+        if count < popular_genre.count(movie_data["genre"]):
+            count = popular_genre.count(movie_data["genre"])
+    return popular_genre[0]
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    watched_movies = user_data["watched"]
+    friends_list = user_data["friends"]
+    user_watched = []
+    dictionary_friend_movies = []

```

Comment:
Since this structure is a list, it's a little misleading to name it starting with `dictionary_` and might cause confusion for people trying to reason about what the variable holds. Since this list will hold all of the friends' movies, I think something like `friends_movies` would be descriptive enough.